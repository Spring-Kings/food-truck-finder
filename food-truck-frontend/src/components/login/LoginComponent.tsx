import React from 'react'
import Form from "../Form";
import {AxiosResponse} from 'axios';
import {Grid, TextField, Typography} from '@material-ui/core'
import {UserData} from "../../redux/user/UserReducer";
import NextLink from 'next/link'
import LinkButton from "../layout/LinkButton";
import Router from "next/router";

type State = {
  resultText: string;
  data: UserData;
  loadUserFromBackend: () => Promise<void>;
}

interface LoginProps {
  data: UserData;
  loadUserFromBackend: () => Promise<void>;
}

class LoginComponent extends React.Component<LoginProps, State> {
  constructor(props: LoginProps) {
    super(props);
    this.state = {
      resultText: "",
      data: props.data,
      loadUserFromBackend: props.loadUserFromBackend
    }
  }

  render() {
    return (
      <Grid container>
        <Grid item>
          <Typography variant="h2">Login</Typography>
        </Grid>
        <Grid item>
          <Form submitUrl={'/login'} onSuccessfulSubmit={this.onSubmit} onFailedSubmit={this.onFail}>
            <TextField label="Username" name="username"/>
            <TextField label="Password" name="password" type="password"/>
          </Form>
        </Grid>
        <Grid item>
          <LinkButton url="/register" text="Create an account here."/>
        </Grid>
        <Grid item>
          <Typography variant="h6">{this.state.resultText}</Typography>
        </Grid>
      </Grid>
    )
  }

  onSubmit = (formData: any, response: AxiosResponse<any>) => {
    const token = response.headers['token'];
    localStorage.setItem('authToken', token);
    this.setState({
      ...this.state,
      resultText: `Set token to ${token}`
    });
    this.state.loadUserFromBackend().catch(err => console.log(err));
    Router.replace('/');
  }
  // TODO: change this once we find the type for the error response
  onFail = (formData: any, response: any) => {
    if (response.response.status === 401)
      this.setState({resultText: "Incorrect username or password."});
    else
      this.setState({resultText: `Failed to login: ${JSON.stringify(response)}`});
  }
}

export default LoginComponent;