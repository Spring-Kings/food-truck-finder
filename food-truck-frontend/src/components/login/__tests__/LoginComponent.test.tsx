import React from "react";
import user_data from "../../../util/test_util/user_data.json";
import { SNAPSHOT_TEST } from "../../../util/test_util/basic_tests";
import LoginComponent from "../LoginComponent";

jest.mock("../../../util/api");

SNAPSHOT_TEST(
  "Snapshot test login component",
  () => {
    return (<LoginComponent data={user_data} loadUserFromBackend={() => Promise.resolve()} />);
  }
);