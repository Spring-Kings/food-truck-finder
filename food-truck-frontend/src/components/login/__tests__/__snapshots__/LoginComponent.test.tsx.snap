// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test login component Matches snapshot 1`] = `"<div class=\\"MuiGrid-root MuiGrid-container\\"><div class=\\"MuiGrid-root MuiGrid-item\\"><h2 class=\\"MuiTypography-root MuiTypography-h2\\">Login</h2></div><div class=\\"MuiGrid-root MuiGrid-item\\"><form><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2\\"><div class=\\"MuiGrid-root MuiGrid-item\\"><div class=\\"MuiFormControl-root MuiTextField-root\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated\\" data-shrink=\\"false\\">Username</label><div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl\\"><input type=\\"text\\" aria-invalid=\\"false\\" name=\\"username\\" value=\\"\\" class=\\"MuiInputBase-input MuiInput-input\\"/></div></div></div><div class=\\"MuiGrid-root MuiGrid-item\\"><div class=\\"MuiFormControl-root MuiTextField-root\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated\\" data-shrink=\\"false\\">Password</label><div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl\\"><input type=\\"password\\" aria-invalid=\\"false\\" name=\\"password\\" value=\\"\\" class=\\"MuiInputBase-input MuiInput-input\\"/></div></div></div><div class=\\"MuiGrid-root MuiGrid-item\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text\\" tabindex=\\"0\\" type=\\"submit\\"><span class=\\"MuiButton-label\\">Submit</span></button></div></div></form></div><div class=\\"MuiGrid-root MuiGrid-item\\"><li class=\\"MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button\\" tabindex=\\"-1\\" role=\\"menuitem\\" aria-disabled=\\"false\\">Create an account here.</li></div><div class=\\"MuiGrid-root MuiGrid-item\\"><h6 class=\\"MuiTypography-root MuiTypography-h6\\"></h6></div></div>"`;
